python/console/console_base.py

```diff
@@ -0,0 +1,174 @@
+
+    def __init__(self, parent=None):
+        super(QgsQsciScintillaBase, self).__init__(parent)
+
+        # Enable non-ascii chars
+        self.setUtf8(True)
+
+        # Set the default font
+        font = QFontDatabase.systemFont(QFontDatabase.FixedFont)
+        self.setFont(font)
+        self.setMarginsFont(font)
+
+        # Margin 0 is used for line numbers (editor and output)
+        self.setMarginWidth(0, "00000")
+        # Margin 1 is used for the '>>>' prompt (console input)
+        self.setMarginWidth(1, "0")
+        self.setMarginType(1, 5)  # TextMarginRightJustified=5
+        # Margin 2 is used for the 'folding' (editor)
+        self.setMarginWidth(2, "0")
+
+        self.setCaretLineVisible(True)
+        self.setCaretWidth(2)
+
+        self.iconRun = QgsApplication.getThemeIcon("console/mIconRunConsole.svg")
+        self.iconRunScript = QgsApplication.getThemeIcon("mActionStart.svg")
+        self.iconUndo = QgsApplication.getThemeIcon("mActionUndo.svg")
+        self.iconRedo = QgsApplication.getThemeIcon("mActionRedo.svg")
+        self.iconCodePad = QgsApplication.getThemeIcon("console/iconCodepadConsole.svg")
+        self.iconCommentEditor = QgsApplication.getThemeIcon("console/iconCommentEditorConsole.svg")
+        self.iconUncommentEditor = QgsApplication.getThemeIcon("console/iconUncommentEditorConsole.svg")
+        self.iconSettings = QgsApplication.getThemeIcon("console/iconSettingsConsole.svg")
+        self.iconFind = QgsApplication.getThemeIcon("console/iconSearchEditorConsole.svg")
+        self.iconSyntaxCk = QgsApplication.getThemeIcon("console/iconSyntaxErrorConsole.svg")
+        self.iconObjInsp = QgsApplication.getThemeIcon("console/iconClassBrowserConsole.svg")
+        self.iconCut = QgsApplication.getThemeIcon("mActionEditCut.svg")
+        self.iconCopy = QgsApplication.getThemeIcon("mActionEditCopy.svg")
+        self.iconPaste = QgsApplication.getThemeIcon("mActionEditPaste.svg")
+        self.iconClear = QgsApplication.getThemeIcon("console/iconClearConsole.svg")
+        self.iconHideTool = QgsApplication.getThemeIcon("console/iconHideToolConsole.svg")
+        self.iconShowEditor = QgsApplication.getThemeIcon("console/iconShowEditorConsole.svg")
+        self.iconPyQGISHelp = QgsApplication.getThemeIcon("console/iconHelpConsole.svg")
+
+    def setLexers(self):
+        self.lexer = QsciLexerPython()
+        self.lexer.setIndentationWarning(QsciLexerPython.Inconsistent)
+        self.lexer.setFoldComments(True)
+        self.lexer.setFoldQuotes(True)
+
+        font = QFontDatabase.systemFont(QFontDatabase.FixedFont)
+
+        # output and console
+        loadFont = self.settings.value("pythonConsole/fontfamilytext")
+        if loadFont:
+            font.setFamily(loadFont)
+        fontSize = self.settings.value("pythonConsole/fontsize", type=int)
+        if fontSize:
+            font.setPointSize(fontSize)
+
+        # TODO: editor has it's own font and size...
```

Comment:
What are we doing here, getting rid of it as we're trying to remove the dual settings situation?

Reply:
Thanks @nirvn Mathieu

I will remove the TODO it is not valid anymore because the editor does NOT have different settings anymore...

And I will rename QgsQsciScintillaBase to QgsPythonConsoleBase as Salvatore proposed